### PR TITLE
Updating and adding favicons for a better UX

### PR DESCRIPTION
**Pull request by Arhell**
https://github.com/nunit/docs/issues/612
Updating and adding favicons for a better user experience

### CODE REVIEW

**Comment by SeanKilleen** [comment]
@Arhell I spun up a Codespace and served the content so that I could look at both URLs and compare them.

| Icon type | Current | With This PR | 
| --------- | -------- | ------------- |
| iOS Safari | ❌ | ✅ |
| Android Chrome | ❌ | ✅ |
| Mac OS X El Capitan Safari | ❌ | ✅ |
| Windows 8 and 10 | ❌ | ❌ | 
| Classic, desktop browsers | ✅ | ❌ |

A few thoughts:

* Do you believe this resolves #612? If so, I recommend updating your PR description to read `Resolves #612`. That way GitHub would automatically close the issue when this is merged. Otherwise, if you'd like to do multiple PRs, I recommend updating the description to `Supports #612`, so that the PR will be linked to the issue but not close it.
* It seems Windows 8 and 10 are unaffected by this PR. Is that by design?
* I'm a little concerned that classic desktop browsers are listed as not working with this PR which would seem to be a regression. Is that your intent?

Thanks for putting in work on this! It'll definitely result in a nicer experience. I'll do an official approval once we walk through the above. Thanks!

**Comment by Arhell** [comment]
Updated icons, unfortunately I cannot check locally

**Comment by SeanKilleen** [comment]
Going to review now -- FYI, the steps I will take:

* I'm opening a GitHub Codespace within GitHub by going to the Code button and clicking on a codespace.
* This is opening a new VS Code instance in the web browser
* I check out the branch
* I open a terminal window and run serve , which is a shortcut command I created that builds the site and serves it up at a Codespaces URL
* I'm then copying that URL into the favicon site you mentioned in order to compare the two.

**Comment by SeanKilleen** [comment]
@Arhell feedback -- 

* When I previewed the site, I again see that the desktop variant appears to fail, saying that all icon sizes are not available. Is it possible this is related to `favicon.ico` being deleted in this PR?
* Please update the description to either `Resolves #612` if the issue is resolved by this PR, or `Supports #612` if it's only one piece of it.

I don't mind merging without all the improvements, but currently it appears this PR introduces a regression where the favicon is no longer considered complete for desktop browsers, which are a large % of our audience.

**Comment by Arhell** [comment]
> @Arhell feedback --
> 
> * When I previewed the site, I again see that the desktop variant appears to fail, saying that all icon sizes are not available. Is it possible this is related to `favicon.ico` being deleted in this PR?
> * Please update the description to either `Resolves #612` if the issue is resolved by this PR, or `Supports #612` if it's only one piece of it.
> 
> I don't mind merging without all the improvements, but currently it appears this PR introduces a regression where the favicon is no longer considered complete for desktop browsers, which are a large % of our audience.

Thanks
returned the old favicon.ico, in any case I will correct it as I see the result

**Comment by SeanKilleen** [comment]
@Arhell I spent some time on this and figured out what the issue is. 

ℹ️ FYI, instead of using codespaces, I was able to use docfx on my local machine to build and serve the site, and then use ngrok.io to forward my local port so that I could run it through the favicon checker.

The issue is that these changes are being made in the `custom_template` folder, which I don't believe is the correct place for such changes, which are largely only for specific overrides (docfx does _not_ make it easy to figure this out.)

I believe if we do the following, everything will work as expected:

* Move the `favicons` folder into the `docs` folder
* Move the favicon.ico and logo.svg into the `docs` folder
* In the `docfx.json` file, in the files to include, add a `favicons/**` entry alongside images etc.

Then, this can be checked by:

* Installing docfx and ngrok (if you use chocolatey, it's `choco install docfx ngrok` to install both.)
* Running `docfx build`
* Running `docfx serve`
* Running `ngrok http 8080 -host-header="localhost:8080"` 
* Placing the ngrok URL plus `/_site` into the favicon checker.

Alternatively, you can check by:

* At the top of this PR, clicking `Code` and then creating a new Codespace:

> image

* Allowing the codespace to load in your browser
* Making the changes I mention
* Running `serve` in the terminal
* Clicking `Browse...` in the bottom-right pop-up that says the site is running (which opens a browser window)
* Placing that URL in the favicon checker.

**Comment by Arhell** [comment]
@SeanKilleen done
But the icon and logo cannot be removed from the custom template folder because docfx generates its own icons and logo

**Comment by SeanKilleen** [comment]
@Arhell thanks! I'll review when I'm able.

I'm yummy meantime, please also update the description of the PR to make it clear how you intend to link to the issue, per my other feedback. 👍

File: docs/custom_template/partials/head.tmpl.partial [hunk 1]
```diff
@@ -16,7 +16,17 @@
   <meta name="title" content="{{#title}}{{title}}{{/title}}{{^title}}{{>partials/title}}{{/title}} {{#_appTitle}}| {{_appTitle}} {{/_appTitle}}">
   <meta name="generator" content="docfx {{_docfxVersion}}">
   {{#_description}}<meta name="description" content="{{_description}}">{{/_description}}
-  <link rel="shortcut icon" href="{{_rel}}{{{_appFaviconPath}}}{{^_appFaviconPath}}favicon.ico{{/_appFaviconPath}}">
+
+  <link rel="apple-touch-icon" sizes="180x180" href="{{_rel}}{{{_appFaviconPath}}}{{^_appFaviconPath}}favicons/apple-touch-icon.png{{/_appFaviconPath}}">
```
Comment (SeanKilleen) [inline docs/custom_template/partials/head.tmpl.partial:20]: 👍 Nice technique here! Taught me something.

File: docs/favicons/manifest.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "docs.nunit.org",
+    "short_name": "nunit docs",
```
Comment (SeanKilleen) [inline docs/favicons/manifest.json:3]:
```suggestion
    "short_name": "NUnit Docs",
```

⛏️ Not sure if this has to be lower-case. If we have flexibility on casing, NUnit Docs would be better.

**Review by SeanKilleen — APPROVED** [review]
👍 LGTM! Nice work, and thanks for your persistence to make all the icons show up right.

ℹ️ FYI, I updated the description of the PR so that it will close #612 once it is merged.

One Nitpick ( ⛏️ ) comment; feel free to merge my suggestion if it makes sense.

🟢 Feel free to merge the PR when you see fit after reviewing the comments. Thanks!

File: docs/custom_template/partials/head.tmpl.partial (hunk 1 above)
Comment (Arhell) [reply docs/custom_template/partials/head.tmpl.partial:20]: You are welcome

File: docs/favicons/manifest.json (hunk 2 above)
Comment (Arhell) [reply docs/favicons/manifest.json]: Done

**Pull request merged**